### DISCUSSION TITLE
Stricter cache control policies don't take effect

### DISCUSSION THREAD

**Discussion by websocket98765** [vercel/vercel#4895] [Help] [answered]
I noticed Vercel automatically sets a cache control header on CSS & JS using:

   `cache-control: s-maxage=31536000, immutable`

Updating vercel.json to a stricter policy (like below) does **not** take effect, and Vercel's default policy remains:

```
{
  "routes": [
    {
      "src": ".chunk.css",
      "headers": {
          "Cache-Control": "public, max-age=31536000, s-maxage=31536000, immutable"
      },
      "continue": true
    },
    {
      "src": ".chunk.js",
      "headers": {
          "Cache-Control": "public, max-age=31536000, s-maxage=31536000, immutable"
      },
      "continue": true
    }
  ]
}

```

This downside of the default policy is that 2nd+ views will encounter a 304, but with the policy I'd like to use, the client won't request/re-validate the asset at all.

To verify this, run a site through https://webpagetest.org, click "Cache Static Content" grade, and note CSS & JS will fail using Vercel's default cache policy.

Are my `vercel.json` directives incorrect? If not, I'd like to feature request allowing stricter cache policies.

Thank you!

**Comment by styfle** [comment, marked as answer]
Did you mean to use a wildcard? `(.*)\.chunk\.css`

https://vercel.com/docs/configuration#routes

**Reply by websocket98765** [reply]
Thank you! That was it. I also added a couple headers to improve the webpagetest.org security score:

```
{
  "routes": [
    {
      "src": ".*",
      "headers": {
        "X-Content-Type-Options": "nosniff",
        "X-Frame-Options": "DENY"
      },
      "continue": true
    },
    {
      "src": "(.*).chunk.(css|js)",
      "headers": {
        "Cache-Control": "public, max-age=31536000, s-maxage=31536000, immutable"
      },
      "continue": true
    }
  ]
}
```
